### ISSUE TITLE
Question about time limits for default runs vs grid search runs

### ISSUE THREAD

**Issue by ArnaudDeza** [Opt-Mucca/Adaptive-Cutsel-MILP#2]
Hello,
We are a group of students attempting to recreate some of the results from this paper. We have the following question regarding runtime limits being set for `pyscipopt`

In the `generate_standard_data.py` file on line 37, the time limit is set as follows: `time_limit=int(time_limit / 60) + 2`
We believe that this means that the time limit for runs using the default cut selection parameters is either 12 seconds (when root is False as in `parameters.py`, FULL_SOLVE_TIME_LIMIT is 600) or either 2 seconds (when root is True as in `parameters.py`, ROOT_SOLVE_TIME_LIMIT is 20). 

In the `parameter_sweep.py` file on line 59, the time limit is set as follows: `time_limit=120`
This means that the time limit to test any configuration (out of all the convex combinations) is 120 seconds.

Is this time limit difference intentional, if so what is the reasoning behind this difference?

Thank you very much for the help + Interesting paper!

**Comment by ArnaudDeza** [comment]
Sorry! I realized that there is the SCIP time limit and Slurm time limit

**Comment by Opt-Mucca** [comment]
Hey!

Incredibly glad to hear that this repository is helping you, and that you enjoyed the paper. 

You correctly spotted that there is a SLURM time limit and a SCIP time limit.
 I made a bad design decision to pass time as minutes to `run_python_slurm_job` (which is the SLURM time limit), so that's why I divide by 60. 
The 120, so two hours, which is the time limit in `parameter_sweep.py` is just a sufficiently large value compared to `ROOT_SOLVE_TIME_LIMIT`. In principle the SLURM time limit needs to be just a bit larger than the SCIP time limit to account for model loading and saving / interpretation of results.

**Comment by ArnaudDeza** [comment]
Thanks for your response.

Another quick question: How is relative gap improvement computed for instances that get solved to optimality when using default cut selection parameters and when using oracle grid search parameters. For example, instance `nexp-150-20-1-5` gets solved to optimality in 11 seconds on our machine using default parameters.

In table 8 from the appendix, this instance has a relative gap improvement of 0.25, but the mip-gap from the default run should already be 0 as it solved to optimality.

**Comment by Opt-Mucca** [comment]
Relative gap improvement if the instance solves to optimality (where in standard condition it didn't) would be 1. 

Are you sure that the instance solved to optimality at the root node (without strong branching enabled, and with my parameter set)? In the current uploaded version of the paper I only included root node results, so solving in 11 seconds in the general solver is very much possible and can happen during the data generation process, but ultimately would not be a result that influences analysis for the root node. All instances that solve to optimality at the root should be filtered. 

I just ran the instance again, and it does solve in the root node, but not under the conditions I impose (a cap on the number of separation rounds and cuts that can be added per round). If you are also using the latest version of SCIP it makes a perfect comparison to my results difficult. I also looked at the log files of the paper, and it does have the result as presented. Thanks for actually running the code and providing feedback! Tell me if the answer isn't suitable or there's still some confusion.

**Comment by ArnaudDeza** [comment]
Thanks for the quick response!

We were solving the instance not at the root and therefore we were able to solve in 11 seconds to optimality. Thank you for pointing this out, the results make much more sense to us now.

Final question: For the results presented in the paper, do you only consider root node solves with a time limit of 20 seconds or 600 seconds for SCIP? I assume it is 20 seconds as that is already long enough for root solves and 600 is overkill for just the root node.

**Comment by Opt-Mucca** [comment]
I only accepted instances where the root node processed in less than 20 seconds. This is somewhat arbitrary and excludes a lot of instances, but the amount of computation power needed is already extreme in my opinion.

I used 600 seconds for the solution generation to get the best available primal after that time limit, and maybe for the extension to branch and bound in the current version of the code. (so that's probably also where you saw it)
